### PR TITLE
colbert naming and organizational chagnes

### PR DESCRIPTION
**Pull request by epinzur**
This is a proposed change to the colbert naming scheme before we do the initial release.

I've disliked the long class names we've had like:

```
from ragstack-colbert import CassandraColbertVectorStore, ColbertCassandraRetriever
```

This simplifies to make the Cassandra part assumed (since it is a datastax product)
```
from ragstack-colbert import ColbertVectorStore, ColbertRetriever
```

But I could see the benefit in keeping the more descriptive names too.

### CODE REVIEW

**Review by zzzming — CHANGES_REQUESTED** [review]
I don't like long names either. It looks good, but I think we need keep the `Cassandra` part. One option is to create a folder called cassandra. So that we can still call it the new name ColBERTVectorStore.

**Comment by epinzur** [comment]
new names are:

```
ColbertRetriever
CassandraVectorStore

ColbertLCRetriever
ColbertLIRetriever
```

File: libs/colbert/ragstack_colbert/base_retriever.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
+"""
```
Comment (epinzur) [inline libs/colbert/ragstack_colbert/base_retriever.py:1]: changed all the interfaces to be prefixed with `base_` and renamed the class name too.

File: libs/colbert/ragstack_colbert/base_embedding.py [hunk 2]
```diff
@@ -10,7 +10,7 @@
 from .chunks import EmbeddedChunk
 
 
-class TokenEmbeddings(ABC):
+class BaseEmbedding(ABC):
```
Comment (zzzming) [inline libs/colbert/ragstack_colbert/base_embedding.py:13]: So the class name should not be plural right?

**Review by zzzming — APPROVED** [review]

**Review by nicoloboschi — APPROVED** [review]

File: libs/colbert/ragstack_colbert/base_embedding.py (hunk 2 above)
Comment (epinzur) [reply libs/colbert/ragstack_colbert/base_embedding.py:13]: i think so... i keep going back and forth on this

File: libs/colbert/ragstack_colbert/cassandra_vector_store.py [hunk 3]
```diff
@@ -1,5 +1,5 @@
 """
-This module provides an implementation of the ColbertVectorStore abstract base class, specifically designed
+This module provides an implementation of the BaseVectorStore abstract class, specifically designed
 for use with a Cassandra database backend. It allows for the efficient storage and management of text embeddings
```
Comment (michaelsembwever) [inline libs/colbert/ragstack_colbert/cassandra_vector_store.py:3]: nit: the technology is named Apache Cassandra, and should initially be referenced as such.  subsequent usages can them be shorten to just Cassandra.

this is mostly FYI.

whether you want to do this here, or somewhere like in the README where a user would first come across it, i leave up to you.

**Comment by michaelsembwever** [comment]
nice improvement! makes the separation of technologies it's using clearer.

the comment was made (elsewhere) that… 

> Cassandra part assumed (since it is a datastax product)

while this is nit-picking and most likely taking the meaning of that sentence out of content, it's worth the gentle reminder that Cassandra is not a datastax product, and any indication of such is a legal branding violation. (The product is Apache Cassandra, and it belongs to the ASF.)

**Pull request merged**